addons/popochiu/engine/objects/walkable_area/popochiu_walkable_area.gd

```diff
@@ -32,110 +32,287 @@ extends Node2D
 var map_rid: RID
 ## Used to assign a map in the [NavigationServer2D] to the region RID of the [b]$Perimeter[/b]
 ## child.
-var rid: RID
+var region_rid: RID
+## Property used by [PopochiuRoom]s to activate the map of this area in the [NavigationServer2D]
+## for characters ignoring obstacles.
+var map_rid_no_obstacles: RID
+## Used to assign a map in the [NavigationServer2D] to the region RID of the [b]$Perimeter[/b]
+## child, for characters ignoring obstacles.
+var region_rid_no_obstacles: RID
+
+## Emitted when the enabled flag changes so the room can react (e.g. switch active map).
+signal enabled_changed(value: bool)
+
+# Reference to the perimeter NavigationRegion2D, saved for internal use
+@onready var _perimeter = get_node_or_null("Perimeter")
 
 
 #region Godot ######################################################################################
 func _ready() -> void:
 	add_to_group('walkable_areas')
 
+	if not _perimeter:
+		print("No perimeter found in the walkable area. Please add a NavigationRegion2D child named 'Perimeter'.")
```

Comment:
Maybe use the Popochiu functions to print messages so devs know this is comming from our plugin?